apps/minifront/src/components/header/top-row.tsx

```diff
@@ -11,15 +11,28 @@ import {
 import { Network } from '@penumbra-zone/ui/components/ui/network';
 import { PagePath } from '../metadata/paths';
 import { TabletNavMenu } from './tablet-nav-menu';
+import { useEffect, useState } from 'react';
+import { getChainId } from '../../fetchers/chain-id';
 import { LayoutLoaderResult } from '../layout';
 
 // Infinite-expiry invite link to the #web-ext-feedback channel. Provided by
 // Henry (@hdevalence) and thus tied to his Discord account, so reach out to him
 // if there are any problems with this link.
 const WEB_EXT_FEEDBACK_DISCORD_CHANNEL = 'https://discord.gg/XDNcrhKVwV';
 
+const useChainId = (): string | undefined => {
+  const { isInstalled, isConnected } = useLoaderData() as LayoutLoaderResult;
+  const [chainId, setChainId] = useState<string | undefined>();
+
+  useEffect(() => {
+    if (isInstalled && isConnected) void getChainId().then(setChainId);
+  }, [isInstalled, isConnected]);
```

Comment:
We should be using react-query for async data loading when it's in a component

Reply:
Ah, my bad — I hadn't seen it being used in any recent code, so I assumed it was effectively being deprecated